### PR TITLE
chore(deps): add `@intlify/shared` and `vue-i18n` v11.0.0 to peer dependencies

### LINKED ISSUE #300: unmet peer @intlify/shared@"^9.0.0 || ^10.0.0": found 11.0.1
### Describe the bug

unmet peer @intlify/shared@"^9.0.0 || ^10.0.0": found 11.0.1

image

### System Info

```Shell
mac
pnpm@latest
```

### Used Package Manager

pnpm

### Additional context

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/intlify/vue-i18n-extensions/blob/next/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/intlify/vue-i18n-extensions/blob/next/CONTRIBUTING.md).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### PR DESCRIPTION
**Pull request by naokihaba**
### Description

Fix peer dependency version ranges to include ^11.0.0

Changes:
- Extend peer dependency version ranges:
 - @intlify/shared: "^9.0.0 || ^10.0.0 || ^11.0.0"
 - vue-i18n: "^9.0.0 || ^10.0.0 || ^11.0.0"

fixes #300

### CODE REVIEW

**Comment by userquin** [comment]
shouldn't be added to the peer dependencies too?

**Review by kazupon — CHANGES_REQUESTED** [review]
@NaokiHaba 
Welcome to contribution!

As @userquin mentioned, the root issue of this issue is the need to add `^11.0.0` to `@intlify/shared` in peer dependencies. Could you add it, please? 🙏

**Comment by naokihaba** [comment]
> @NaokiHaba Welcome to contribution!
> 
> As @userquin mentioned, the root issue of this issue is the need to add `^11.0.0` to `@intlify/shared` in peer dependencies. Could you add it, please? 🙏

@kazupon @userquin 

Thank you for your response. I understand. 🙇
I will proceed with creating the pull request. Once the changes are complete, I will notify you again.

**Review by kazupon — COMMENTED** [review]
@NaokiHaba 
Thanks for your updates!
Do I correct in understanding that you updated this snapshot because the version of vue on which vue-i18n-extensions depends changed from 3.4 to 3.5, don’t you?

**Comment by kazupon** [comment]
If possible, I would like to only update `pnpm-lock.yaml` in `@intlify/shared` in this PR. Can you do that?

**Comment by NaokiHaba** [comment]
> If possible, I would like to only update `pnpm-lock.yaml` in `@intlify/shared` in this PR. Can you do that?

Yes, you're correct! The snapshot updates were necessary because of test failures that occurred after the Vue dependency update in vue-i18n-extensions from 3.4 to 3.5.

However, I agree that it would be better to handle these changes separately:

1. First PR (this one): Only update the `pnpm-lock.yaml` for `@intlify/shared`
2. Second PR: Handle the test updates and lint errors

I'll revert the snapshot changes in this PR and create a separate PR for the test updates. This will make the changes clearer and easier to review.

Would you like me to proceed with this approach?

**Comment by kazupon** [comment]
I've noticed that we need `vue-i18n@^v11.0.0` to `peerDependecies` from this issue
https://github.com/intlify/bundle-tools/issues/431

the bellow log from that issue:
```
packages/frontend
└─┬ @intlify/unplugin-vue-i18n 6.0.2
  └─┬ @intlify/vue-i18n-extensions 7.0.0
    ├── ✕ unmet peer @intlify/shared@"^9.0.0 || ^10.0.0": found 11.0.1
    └── ✕ unmet peer vue-i18n@"^9.0.0 || ^10.0.0": found 11.0.1
```

**Comment by naokihaba** [comment]
> @NaokiHaba Thanks for your updates! Do I correct in understanding that you updated this snapshot because the version of vue on which vue-i18n-extensions depends changed from 3.4 to 3.5, don’t you?

Yes, that's exactly right! The snapshot and lint failures occurred because of the Vue dependency update from 3.4 to 3.5 in vue-i18n-extensions.

Here's what I'm seeing:
1. Snapshot test failures (22 failed) showing changes in the AST structure, particularly around `cached` and `patchFlag` properties
2. TypeScript/ESLint errors related to unsafe Function calls across multiple files

I chose to update these tests as they were failing after the dependency update. 

> If possible, I would like to only update `pnpm-lock.yaml` in `@intlify/shared` in this PR. Can you do that?

Of course, that's possible! I'll focus only on updating the `pnpm-lock.yaml` in `@intlify/shared` for this PR.

**Comment by kazupon** [comment]
I apologize for the lack of context in my approach to this issue.
The Peer dependecies issue should be resolved by simply updating `peerDependecies` without updating the `dependecies` version.
So, it should be possible to resolve it only by updating the following.

```diff
diff --git a/package.json b/package.json
index a11c3f1..e75e121 100644
--- a/package.json
+++ b/package.json
@@ -59,10 +59,10 @@
     "vue": "^3.2.45"
   },
   "peerDependencies": {
-    "@intlify/shared": "^9.0.0 || ^10.0.0",
+    "@intlify/shared": "^9.0.0 || ^10.0.0 || ^11.0.0",
     "@vue/compiler-dom": "^3.0.0",
     "vue": "^3.0.0",
-    "vue-i18n": "^9.0.0 || ^10.0.0"
+    "vue-i18n": "^9.0.0 || ^10.0.0 || ^11.0.0"
   },
   "peerDependenciesMeta": {
     "@intlify/shared": {
```

**Comment by naokihaba** [comment]
> I apologize for the lack of context in my approach to this issue. The Peer dependecies issue should be resolved by simply updating `peerDependecies` without updating the `dependecies` version. So, it should be possible to resolve it only by updating the following.

Thanks for the clarification!

I'll update both  `vue-i18n` in the `peerDependencies` field accordingly.

**Comment by kazupon** [comment]
Ah, You need to revert `dependecies` field (revert https://github.com/intlify/vue-i18n-extensions/pull/301/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R24) and `pnpm-lock.yaml`

**Comment by naokihaba** [comment]
> Ah, You need to revert `dependecies` field (revert https://github.com/intlify/vue-i18n-extensions/pull/301/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R24) and `pnpm-lock.yaml`

I've made the requested changes:
1. Reverted the `dependencies` field changes from PR #301
2. Reverted the `pnpm-lock.yaml` changes

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -1293,6 +1293,7 @@ packages:
   /@intlify/shared@11.0.1:
```
Comment (kazupon) [inline pnpm-lock.yaml:1293]: Hmm, `pnpm-lock.yaml` has not reverted completly...
You can try again with a new PR, committing only the following changes.

```diff
diff --git a/package.json b/package.json
index a11c3f1..e75e121 100644
--- a/package.json
+++ b/package.json
@@ -59,10 +59,10 @@
     "vue": "^3.2.45"
   },
   "peerDependencies": {
-    "@intlify/shared": "^9.0.0 || ^10.0.0",
+    "@intlify/shared": "^9.0.0 || ^10.0.0 || ^11.0.0",
     "@vue/compiler-dom": "^3.0.0",
     "vue": "^3.0.0",
-    "vue-i18n": "^9.0.0 || ^10.0.0"
+    "vue-i18n": "^9.0.0 || ^10.0.0 || ^11.0.0"
   },
   "peerDependenciesMeta": {
     "@intlify/shared": {
```

Comment (naokihaba) [reply pnpm-lock.yaml:1293]: I apologize for the incomplete revert. I'll create a new PR that includes only the necessary `pnpm-lock.yaml` changes. 🙇

**Pull request closed without merging**